pkg/telemetry/phonehome/interceptor.go

```diff
@@ -0,0 +1,65 @@
+package phonehome
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	erroxGRPC "github.com/stackrox/rox/pkg/errox/grpc"
+	"github.com/stackrox/rox/pkg/grpc/authn"
+	grpcError "github.com/stackrox/rox/pkg/grpc/errors"
+	"github.com/stackrox/rox/pkg/grpc/requestinfo"
+	"github.com/stackrox/rox/pkg/sync"
+	"google.golang.org/grpc"
+)
+
+var (
+	mux = &sync.Mutex{}
```

Comment:
I find it hard to track vars defined in one file and used in other files. IIUC, this is only used to guard write access to `Config.interceptors`. If so, then I think:
* it shall at least be declared in `client_config.go`
* but actually it should be part of `Config`: `Config.interceptorsLock` because otherwise a single mutex is used for all `Config`s (I know we have a singleton but this is not clear here and hence breaks the abstraction)
* `track()` should likely copy `cfg.interceptors` under the same mutex
* probably should be a `sync.RWMutex`